### PR TITLE
Thorium High Temperature Reactor Rework

### LINKED ISSUE #8063: [bartworks] Thorium High Temperature Reactor issues
#### Which modpack version are you using?
2.1.0.0 (also tested with 2.1.07a)
#
#### If in multiplayer; On which server does this happen? 
SP
#
#### What did you try to do, and what did you expect to happen?
Build and run the Thorium High Temperature Reactor from bartworks to eventually obtain Lutetium in large quantities
#
#### What happened instead? (Attach screenshots if needed)
I found the following issues:

Structure check is successfully passed even though an output bus is needed (also not mentioned in the tooltip)

IC2 coolant input is possible, tricorder shows a conversion ratio from coolant to hot coolant, but no hot coolant is output (the output hatch in the below image was set to output hot coolant via screwdriver and faced the quantum tank)
grafik

Spent fuel pebbles are ejected into the inventory of the output hatches
grafik
grafik

In **emptying mode**, the fuel pebbles are both ejected into the output bus as well as the output hatches
grafik

#
#### What do you suggest instead/what changes do you propose?
Having briefly spoken to bart on discord, he mentioned that changes to base GT may be the cause of this
#
#### What is your GTNH Discord username?
Afx237v7

### PR DESCRIPTION
**Pull request by kuba6000**
### Thorium High Temperature Reactor Improved/fixed/changed
- RIP BISO
- Efficiency is now **shown** and calculated **exponentially** based on real life ((fuel-100000)/57500)^2+1 (max 100)
- Takes 5% fuel multiplied by efficiency per operation
- Reactor will now stop when coolant is not supplied (not 100% needed)
- Hot coolant is now output constantly insted of at the end of recipe
- Fuel recipes not changed 
- Removed Lutetium from recycling, now only recycles Lead
- Now reactor uses power to operate (0.5A IV)
image
image

### [RECIPES](https://imgur.com/a/ZkG1Uov)
- Based on THTR
- Efficiency is **shown** and calculated **exponentially** based on real life (fuel/72000)^2
- Takes 5% fuel multiplied by efficiency per operation
- Max capacity 720000
- Min capacity 10%
- Uses 1A LUV to operate
- 20x less coolant usage
- Only **one** fuel can be active at a time
- Fuels:
  1. TRISO (Thorium) -> Lutetium   (Thorium232 + Uranium235)
  2. TRISO (Uranium) -> Krypton    (Uranium238 + Uranium235)
  3. TRISO (Plutonium) -> Xenon   (Plutonium239 + Plutonium241)
- Improved emptying mode so it is not voiding items

### New controller description
image
image

### Tooltip for fuels
image
Coolant multipliers:

- Thorium 1x
- Uranium 1.5x
- Plutonium 2x

### New textures
image

### TODO

- ~~Structure still forms only with maintenance hatch [and...](https://imgur.com/a/4idKPJ7)~~ fixed
- ~~Description~~ fixed
- ~~Energy/coolant rebalancing~~ fixed?

Fixed https://github.com/GTNewHorizons/GT-New-Horizons-Modpack/issues/8063

In regards to https://github.com/GTNewHorizons/GT-New-Horizons-Modpack/issues/7916

- This pull request adds a viable way to obtain lutetium starting at LUV/ZPM

### I summarised all balance changes in this chart // OLD Only first fuel
image

### TL;DR
Lutetium goes **FAST**, prices go **UP** and machine **WORKS**

### CODE REVIEW

**Comment by kuba6000** [comment]
@Dream-Master What do you think about this?

**Comment by Dream-Master** [comment]
the changes looks cool . Ok i not tested ingame. We need more thoughts from others more advanced players have much more experience than me. i just make the pack and developed it. I never had the time to play it extensive.

**Comment by GTNH-Afx237v7** [comment]
As mentioned on discord, I can’t judge the code but it looks promising from the description. I am all for putting this in dev and giving it a try.

**Comment by Savallator** [comment]
Since it is a High-Temperature Reactor: How about adding a new fluid "superheated Coolant",and add a new LHE recipe to turn it into hot coolant? That way it can output massive amounts of heat without outputting massive amounts of fluids, and it would make the setup a bit more interesting.
Edit: Also, i am just so close to starting one up in my world, so it is not true noone built them lol

**Comment by Pxx500** [comment]
> Since it is a High-Temperature Reactor: How about adding a new fluid "superheated Coolant",and add a new LHE recipe to turn it into hot coolant? That way it can output massive amounts of heat without outputting massive amounts of fluids, and it would make the setup a bit more interesting.
> Edit: Also, i am just so close to starting one up in my world, so it is not true noone built them lol

But what is supposed to be the output from the LHE? Superheated steam has very low energy density compared to plasma (since this reactor is LuV/ZPM)

**Comment by Savallator** [comment]
The Initial draft is Hot Coolant (So Superheated Steam) As well...the output of the LHE would be superheated steam then, and then the output coolant could go to another LHE for more superheated steam

**Comment by Savallator** [comment]
It would also be nice if we could get some turbine Materials with it, like Steam-optimized turbines with very high flow rate and good Durability for steam but significantly worse Efficiency.

**Comment by Pxx500** [comment]
> It would also be nice if we could get some turbine Materials with it, like Steam-optimized turbines with very high flow rate and good Durability for steam but significantly worse Efficiency.

That's what huge loose turbines are for I guess

**Comment by Savallator** [comment]
I am not sure if the loose mode actually works, but if it does, then yes. But that is kind of a different Issue anyways

**Comment by Savallator** [comment]
Also, the reactor could in theory be built in IV in it's current state.
So not LuV/ZPM.

**Comment by Pxx500** [comment]
> Also, the reactor could in theory be built in IV in it's current state.
> So not LuV/ZPM.

I know it can be done in IV but if there is anyone that will do that then I would assume there is something wrong with them

**Comment by Savallator** [comment]
Ehm...

**Comment by Savallator** [comment]
it is not that hard really. In it's current state, the hard part is how to buffer enough cold coolant, and how to process all the steam, and what to do with the Energy...

**Comment by Pxx500** [comment]
> it is not that hard really. In it's current state, the hard part is how to buffer enough cold coolant, and how to process all the steam, and what to do with the Energy...

This PR makes it more than 10x harder

**Comment by Savallator** [comment]
Well, i did not look at the reactors craft recipe since i have no test setup here to actually launch the PR. But from the first post, it looks not that much harder.
Edit: And looking at the source, the crafting recipe did not change. How exactly is it harder to make then?
You will need some more Thorium initially, but since the minimum amount of pebbles decreased, it is nowhere 10x, and Thorium and Uranium comes in fast with MBM.

**Comment by kuba6000** [comment]
> Well, i did not look at the reactors craft recipe since i have no test setup here to actually launch the PR. But from the first post, it looks not that much harder.
> Edit: And looking at the source, the crafting recipe did not change. How exactly is it harder to make then?
> You will need some more Thorium initially, but since the minimum amount of pebbles decreased, it is nowhere 10x, and Thorium and Uranium comes in fast with MBM.

At peak efficiency you need 900 Th232 per hour, I dont think you can make that much easily at IV and to fill the reactor to 100% you need 180k Th232 (240k Th) that is ~144ML of fluorine

**Comment by Savallator** [comment]
But
1. you need only 10% to start it up
2. you don't need to run it continiously

**Comment by Savallator** [comment]
Also, if it gets pushed further to the lategame, do you really think people would use it more?
I don't think so, especially if the alternatives are plasma based .
And in real life, pebble bed reactors are a pretty safe and comparably "low-tech" design.

**Comment by Prometheus0000** [comment]
Suggestion: remove coolant to make it clear it's not for power. Or just have people understand that.

**Comment by Savallator** [comment]
Imho is is a great power option to fill the gap between Fluid reactors and Fusion... The balance just needs to be tuned.

**Comment by Prometheus0000** [comment]
I'd rather it exist just for the products it produces, otherwise balancing between the two might be challenging.

**Comment by Savallator** [comment]
Suggestion: reverse the Efficiency, so it gets less efficient with more pebbles, but while partially filled, it generates less Lutetium. That way it can be used for IV/LuV tier power, and when used to produce Lutetium you should fully fill it, without the Power going through the roof. That could be done by having a chance to return an "empty" pebble without Lutetium in it (just the Silicon/Carbon), and that chance depends on amount of pebbles.

**Comment by Savallator** [comment]
So, if you fire it up for power with just 10% pebbles (or potentially less), it generates Lutetium only from every 10th pebble. That way, to get the products, you need to fill it fully, but for Power it gets viable earlier.

**Comment by Pxx500** [comment]
> I'd rather it exist just for the products it produces, otherwise balancing between the two might be challenging.

The heat it produces is supposed to be more of an inconvenience than advantage since you can't just delete coolant.

**Comment by Prometheus0000** [comment]
? You can though? Fluid trash can.

**Comment by Pxx500** [comment]
> ? You can though? Fluid trash can.

You have to mine quite a lot of lapis

**Comment by Savallator** [comment]
But you can not make hot coolant to cold coolant for free, if you use a trashcan, you need a steady supply of new coolant.

**Comment by Prometheus0000** [comment]
I don't disagree. I'm just saying you _can_ void it.

**Comment by Pxx500** [comment]
Technically yes

**Comment by Savallator** [comment]
actually, you can make coolant from hot coolant. Make a LHE and never supply any water. Iirc it will not explode unless a hot LHE gets water, and it should still cool the coolant.

**Comment by Pxx500** [comment]
And it may not seem like it but I tried to keep this reactor as real as possible, both byproducts (krypton and xenon) are technically present in the decay chains of uranium and plutonium.

**Comment by Savallator** [comment]
But what do you think of my idea to reverse efficiency and let producst scale with pebble amount? It even makes a bit sense from a real world view.

**Comment by Pxx500** [comment]
It is the exact opposite of how real world works, it should scale exponentially both in terms of heat and products

**Comment by Savallator** [comment]
A bigger reactor would drop in Efficiency due to more overhead with the cooling, and a smaller would (in theory, but the effect is minimal i guess) produce less products due to less Neutrons.

**Comment by Pxx500** [comment]
> A bigger reactor would drop in Efficiency due to more overhead with the cooling, and it would (in theory, but the effect is minimal i guess) produce less products due to less Neutrons.

That's why we cool it to keep it efficient

**Comment by Savallator** [comment]
and the heat will not scale exponentially in any way... the amount of generated heat depends of the number of fissions per timeframe, and that slows down if the heat can not be removed fast enough, and that gets hard to do with a large pebble stack. So a larger reactor would drop a bit in efficiency.

**Comment by Pxx500** [comment]
> and the heat will not scale exponentially in any way... the amount of generated heat depends of the number of fissions per timeframe, and that slows down if the heat can not be removed fast enough, and that gets hard to do with a large pebble stack. So a larger reactor would drop a bit in efficiency.

That's the part in which we ignore real world limitations because gregtech does it everywhere (more than 100% efficiency in multiple generators).

**Comment by Savallator** [comment]
But still, it would make sense to have heat conversion efficiency drop with more pebbles. I mean, it will make more still, but the curve should flatten a bit.

**Comment by Pxx500** [comment]
How about the reactor being DESIGNED TO HANDLE THAT MUCH?

**Comment by Savallator** [comment]
And that way it would be nice to build earlier for the power, even though it is still a challenge, and later on you could convert your setup to make Lutetium. Also, it would then be nice to have a coolant tower multiblock to just turn hot coolant into coolant in the lategame if you don't care for the power.

Edit: Actually i like the idea of a coolant tower. If someone is good at making textures, i could make a draft for the code.

**Comment by Pxx500** [comment]
> And that way it would be nice to build earlier for the power, even though it is still a challenge, and later on you could convert your setup to make Lutetium. Also, it would then be nice to have a coolant tower multiblock to just turn hot coolant into coolant in the lategame if you don't care for the power.
> 
> Edit: Actually i like the idea of a coolant tower. If someone is good at making textures, i could make a draft for the code.

We've done some calculations and generating large amounts of power (100-300kEU/t  since you want to fill the Fluid Nuke/ Fusion gap) through steam/HP steam is not feasible due to it's very small energy density. Doing so would require a large infrastructure with multiple LHEs and a lot of turbines due to small hatch size at these tiers (even later tiers are not that much better at that). If you have a better idea on how reactor could generate more electricity (60k atm) while also not spamming machines I would be happy to hear about it.

**Comment by Savallator** [comment]
It would need some more changes, but how about a third steam tier along with it, supercritical steam? It could make double the EU of Superheated and output Superheated. That would simplify transport, as its energy density would increase by 250% compared to superheated. The THTR being an Reactor designed for higher operating temperatures would then be able to make superheated coolant, and that would make Supercritical steam in the LHE (and cool down to coolant). Then you could chain 3 turbines, which is some infrastructure, but since the first generates 4 times the amount of a normal steam turbine, it generates more energy per turbine overall. The Multiamp dynamo hatches will work nice with that. 
Edit:
This would be real-world inspired: https://en.wikipedia.org/wiki/Supercritical_steam_generator

**Comment by Prometheus0000** [comment]
We don't need more steam, it's bloody steam.

**Comment by Savallator** [comment]
But it would have a massively increased energy density.
I mean, in the real world almost all power that comes from heat is steam based...

**Comment by Prometheus0000** [comment]
*Video game*

We could make a black hole generator, or a interdimensional power thing or w/e. But *steam*, you want more *steam*, really?

**Comment by Savallator** [comment]
That is why i proposed supercritical steam, with a higher energy density. I was not the one in this discussion who brought up real world realism by the way.

**Comment by Savallator** [comment]
If it should be something other than steam, it is either just another liquid concept, like hot liquid Sodium, going directly into a special turbine and cycling back, or something more exotic. Maybe using the reactor to output some fluid usable in some kind of fuel cell... I was thinking hydrogen at first (reactor input water, output hydrogen), but the energy density would suck.

**Comment by Savallator** [comment]
If you want to go crazy, the reactor could also output some highly radioactive fluid mixture (in a way like the newly added Naquadah fuels, just more low-tech), and that would be fed into another RTG-Style Multi, that outputs Energy (maybe with water coolant or even boosted with cryotheum)

**Comment by Prometheus0000** [comment]
I don't want it to produce power to begin with.

> The heat it produces is supposed to be more of an inconvenience than advantage since you can't just delete coolant.

I see nothing wrong with this. You get a little power out of it I guess, the end. If it produces too much power, add two new coolant fluids, and have it cooled in the VF. Or use the cells.

**Comment by Savallator** [comment]
But i am quite sure you can actually cool down coolant without turbines and stuff, by using a LHE without water.
Also, i was a bit confused initially by using a pebble bed reactor just for Isotope generation, as the real world design is not really invented for breeding stuff (since it is very difficult to get anything out of used pebbles), but specifically to use otherwise difficult Thorium to produce power. If it is really just for Lutetium Generation, why not push it further to the endgame, at least post-fusion?
Maybe a type of Molten Salt Pebble bed reactor would then be nice to introduce for generating power. It could share a part of the recipes, but sacrifice Isotope generation efficiency for power production. Should make around 150k EU/t or something (like two times as much as a Fluid-reactor) and be very Efficient. That way the existing HTGR could stay for Isotope generation.
Edit: I get your point that it is difficult to balance Power generation and breeding in one reactor...

**Comment by Prometheus0000** [comment]
> If it is really just for Lutetium Generation, why not push it further to the endgame, at least post-fusion?

Sounds good to me.

**Comment by bartimaeusnek** [comment]
That thing is litterally made with real values from the real THTR that was build in Munster, not balanced or anything, more or less a reference to the original thing and I would be quite sad if you just replace it instead of just doing a new Multiblock instead.

**Comment by Prometheus0000** [comment]
This is a video game though. It doesn't need to have realistic values to reference something. Better to have it be actually useful, so it actually gets used. Otherwise no one's even going to see the reference in the first place.

**Comment by Houstonruss** [comment]
I suggested this the day bart showed it to me so good to see someone else wanted that reactor to be like this.  This solves the lutecium problem and coolant problem so I'm happy.

**Comment by basdxz** [comment]
Bart's point is to make this a new multiblock which is something I agree with. This is not a rework, this is a straight up replacement of one design for an entirely different one. It would make much more sense to make Bart's design get the flow changes so the coolant output is gradual while making this new design a separate, different multiblock.

**Comment by Pxx500** [comment]
> This is not a rework, this is a straight up replacement of one design for an entirely different one.

How's that? It's the same type of reactor, THTR is just a reactor from HTGR group the only thing we did was to broaden it's possibilities. It works the same way but just has more fuel and different numbers, how is that not a rework?

**Comment by GTNH-Colen** [comment]
I don't see why this shouldn't be merged, this multi is currently completely redundant and unused. @kuba6000 has obviously put many hours into writing a functioning version only for people to get mad that it makes the obsolete one obsolete? What?

**Comment by Savallator** [comment]
> 
> 
> I don't see why this shouldn't be merged, this multi is currently completely redundant and unused. @kuba6000 has obviously put many hours into writing a functioning version only for people to get mad that it makes the obsolete one obsolete? What?

That is bullshit, it is not unused.
I agree that it is a bit overpowered, because it makes massive power (like 1,4m EU/t), but it also needs quite some setup (like 3m lapis just for coolant), so that seems fair.

**Comment by Savallator** [comment]
Also, as i already said, maybe consider making it a different type of reactor, like a Liquid Sodium Reactor or something like that. 
That would fit better to the "generate Isotopes" theme.
Edit:
It could just use the numbers and mechanics OP implemented, but that would also solve the coolant thing: Just make it use Sodium, outputting hot sodium, needing a freezer to cool down...
That way it is clear that you won't get energy, and it needs some infrastructure.

**Comment by Pxx500** [comment]
> That way it is clear that you won't get energy, and it needs some infrastructure.

What's with this making things clear? It's ZPM I'm pretty sure people that got here know basic maths and game balance to understand that 200l/t coolant is pretty much nothing. Throughout the pack quests get less and less detailed, if you really want to say "it's not for power" it would take much less time to do in quest rather than recode the reactor.

**Comment by Savallator** [comment]
The old recipe is nowhere near ZPM, more like EV if you are really determined.
So how about the proposal to just make it a seperate reactor?

**Comment by Pxx500** [comment]
> The old recipe is nowhere near ZPM, more like EV if you are really determined.

Have you seen updates on discord and on this post?
Recipes are 100% ZPM or even higher, not only is it impossible without fusion but it's also VERY resource and energy consuming

**Comment by Savallator** [comment]
But Bart was talking about the old recipe, and said he really did not wanted it removed, like you wanted to do. And i totally agree with him. So why not put your changes into a new Multi and leave the old in place?

**Comment by Pxx500** [comment]
> > I don't see why this shouldn't be merged, this multi is currently completely redundant and unused. @kuba6000 has obviously put many hours into writing a functioning version only for people to get mad that it makes the obsolete one obsolete? What?
> 
> That is bullshit, it is not unused.
> I agree that it is a bit overpowered, because it makes massive power (like 1,4m EU/t), but it also needs quite some setup (like 3m lapis just for coolant), so that seems fair.

How is it not unused? It doesn't even generate hot coolant untill this PR gets merged

**Comment by Savallator** [comment]
I think it is an awesome Multiblock, with huge setup cost, but also huge reward in the form of Fusion-like power, but pre-fusion.

**Comment by Savallator** [comment]
how does it not? it just outputs it all at the end?

**Comment by Pxx500** [comment]
There are multiple more experienced players disagreeing with you. This multi is not only weird in terms of balance (why would it output at the end?) but it also has broken structure. The fact that it makes so much steam that it requires you to make 400 TURBINES TO USE IT ALL makes it questionable at best. The reason why fusion seems so overpowered before you use it is because it is meant to save your game from dying. Making one of those reactors before our changes is not only trivial in EV but it is also stupid because you can just make naq gens to get the energy cheaper and it wouldn't hit your TPS and FPS as much as this reactor infrastructure. Turbines ALONE would cost you so much you would be better to make fusion anyway because it's cheaper.
TL:DR
Reactor bad now

**Comment by Savallator** [comment]
it does not use 400 turbines, and also, the way it is right now you can use fewer turbines, as it does not punish you for only using a fraction of it.

**Comment by Savallator** [comment]
On maximal pebble amount, it generates 4800 coolant per tick, with 60 LHE that can be fed into 120 HP turbines, with the LP turbines that is 240 at most, so not 400. That generates around 2mEU/t. But actually, loose mode is a thing as well.

**Comment by Savallator** [comment]
Also, the "more experienced player" card is a bit arrogant, as you have no idea about my experience with mc, gtnh or whatever. Ad hominem does not make your arguments true...

**Comment by MMunier** [comment]
> 
> 
> There are multiple more experienced players disagreeing with you. This multi is not only weird in terms of balance (why would it output at the end?) but it also has broken structure. The fact that it makes so much steam that it requires you to make 400 TURBINES TO USE IT ALL makes it questionable at best. The reason why fusion seems so overpowered before you use it is because it is meant to save your game from dying. Making one of those reactors before our changes is not only trivial in EV but it is also stupid because you can just make naq gens to get the energy cheaper and it wouldn't hit your TPS and FPS as much as this reactor infrastructure. Turbines ALONE would cost you so much you would be better to make fusion anyway because it's cheaper.
> TL:DR
> Reactor bad now

How do you get to 400 Turbines? 
If you take the efficiency hit for using loose mode you'd max out the reactor by using 12 HP and LP turbines each when using large oriharukon turbines, generating slightly more than 1M EU/t.

**Comment by Pxx500** [comment]
Stop writing multiple comments and make one that you either edit or think about it.



> Also, the "more experienced player" card is a bit arrogant, as you have no idea about my experience with mc, gtnh or whatever. Ad hominem does not make your arguments true...

This is not ad hominem, I'm stating a fact that multiple more experienced players myself excluded disagree with you. I never considered myself experienced, but mods and Devs are mostly on my side and the only one that is on yours doesn't even talk balance wise, he only wants a reference.

240 turbines is still way too much and you don't seem to understand what other Devs intended when making power gen, you should NEVER need to spam THAT much. 240 is still a significant amount.

@MMunier pointing my error twice doesn't make your arguments stronger. Also you wouldn't make that much on loose.

**Comment by Savallator** [comment]
> 
> 
> I don't see why this shouldn't be merged, this multi is currently completely redundant and unused. @kuba6000 has obviously put many hours into writing a functioning version only for people to get mad that it makes the obsolete one obsolete? What?

Also, to answer that: Why remove a Multi that Bart put many hours into to model after the existing THR-300? How do you see it obsolete? The recent changes to TH-232 processing even made it more interesting to use. The rework just removed a major point of the reactor.

**Comment by MMunier** [comment]
@Pxx500 Maybe I'm wrong but at least as far as the tooltip on the Turbine Rotor is concerned, it states loose mode has half the efficiency of regular mode.
Since the regular mode would output 2M EU/t, loose mode should output 1M EU/t.

**Comment by Pxx500** [comment]
> > I don't see why this shouldn't be merged, this multi is currently completely redundant and unused. @kuba6000 has obviously put many hours into writing a functioning version only for people to get mad that it makes the obsolete one obsolete? What?
> 
> Also, to answer that: Why remove a Multi that Bart put many hours into to model after the existing THR-300? How do you see it obsolete? The recent changes to TH-232 processing even made it more interesting to use.

1. It's not only broken but also unfinished
2. No one gets the reference anyways, if you want it to be a reference at least some people understand make it more clear.
3. It's useless (you seem to be the only one that uses it and defends it balance wise) and unbalanced, also doesn't fit into gregtech style of machines having real requirements (doesn't explode without coolant)
4. I was the one to design(not implement) Th232 line with this reactor rework in mind. Also after rework it still usesTh232 so what's your point?

**Comment by Pxx500** [comment]
> @Pxx500 Maybe I'm wrong but at least as far as the tooltip on the Turbine Rotor is concerned, it states loose mode has half the efficiency of regular mode.
> Since the regular mode would output 2M EU/t, loose mode should output 1M EU/t.

It wouldn't output 2M,salvator and Kuba calculated it to be 1,4M

**Comment by Savallator** [comment]
> 
> 
> > @Pxx500 Maybe I'm wrong but at least as far as the tooltip on the Turbine Rotor is concerned, it states loose mode has half the efficiency of regular mode.
> > Since the regular mode would output 2M EU/t, loose mode should output 1M EU/t.
> 
> It wouldn't output 2M,salvator and Kuba calculated it to be 1,4M

> No one gets the reference anyways, if you want it to be a reference at least some people understand make it more clear.

I think the reference is quite clear, but yes, it could be renamed to THTR-300.
That was without turbine efficiency. 1,4M*0,75 is 1M

**Comment by Pxx500** [comment]
> > > @Pxx500 Maybe I'm wrong but at least as far as the tooltip on the Turbine Rotor is concerned, it states loose mode has half the efficiency of regular mode.
> > > Since the regular mode would output 2M EU/t, loose mode should output 1M EU/t.
> > 
> > 
> > It wouldn't output 2M,salvator and Kuba calculated it to be 1,4M
> 
> That was without turbine efficiency. 1,4M*0,75 is 1M

1,4M is at 150% Eff
It also doesn't matter

**Comment by Savallator** [comment]
it is basically 4800(coolant per tick)*200(LHE factor)*1,5(sh steam eu ratio)*0,75(turbine efficiency in loose) = 1080.000

**Comment by Pxx500** [comment]
As I said it doesn't matter, 240 turbines is all that is important for the sake of argument

**Comment by Savallator** [comment]
24 FFS

**Comment by Pxx500** [comment]
@Prometheus0000 we can talk all day but I really don't have the time, could you please tell which one is right or wrong and end it now? If you don't feel that you should do it then I'm sorry for ping but you are the first one that came to my mind

**Comment by Savallator** [comment]
Just to summarize:  with full pebbles, it currently makes 960.000 sh steam/tick, that can be made to 1.080.000eu/t using 12 HP turbines  and 12 LP turbines, with loose mode oriharouken. 24 Turbines is not that much imho.
Also, you can use it with much less than that.

**Comment by MMunier** [comment]
I believe the main reason this Multiblock was unattractive before wasn't the Multi itself, but the practical imposibility to obtain enough Th232 to even be able to run it. Since Thorium-Processing was reworked, we're now able to generate Th232 in large amounts, so this Multi became viable as a power source directly pre Fusion.

**Comment by Prometheus0000** [comment]
Given that you are the only person to even try building it that I know about, it's not really worth the consideration of whether it would mess up existing setups to worry about that.

240 turbines is 10x more than is acceptable for a single power structure in my opinion. At this level, the only reason to make it is to make something super sub-optimal just for the sake of it. 

Regardless of whether it was or was not for power before, it was not effective, and now is not for power.

I understand that you don't want it to change, but that's too bad, sometimes things change. You should appreciate it's new use, and look at the GG mod for the new gen(s) it has.

Bart is not an active dev, so while his input is appreciated, he does not get the final say.

Given that it didn't work properly before, having it work properly now is good. I suggested adding the VM fluids to it as an output to make it more useful. If this makes it less like the 'real thing' that I've never heard of (and probably the same for many people) before, well, that's too bad. Sometimes you have to sacrifice realism for gameplay, like the addition of redstone in vanilla.

**Comment by Savallator** [comment]
It needs 24 Turbines, and that only when you have it full with pebbles and maxed out.

**Comment by Prometheus0000** [comment]
> Regardless of whether it was or was not for power before, it was not effective, and now is not for power.

**Comment by GTNH-Colen** [comment]
Regardless of all the nonsense above I think it's very safe to say that this will be approved for merger once you've finished @kuba6000. Thanks for putting in the effort to revamp this.

-GTNH Team.

**Comment by repo-alt** [comment]
I think making new version of the reactor controller derived from the old one is trivial, and if there is an actual user apperared in the issue, we may safely assume there are more silent ones somewhere out there. I don't think it would be hard for the PR author to do such refactoring and it will decrease objections?
New controller may even use the same texture, etc.

**Comment by Prometheus0000** [comment]
I don't really think it's worth the effort to make another MB for this. Retaining a MB that almost no one uses because it's so bad is silly. If necessary, a second mode could be added (like the cutting factory or chemical washer multis in GT++), if there's enough people who desperately want something greatly sub-optimal.

Of course, that's assuming that there aren't a decent # of users, in which case it would be an ok idea.

**Comment by Dream-Master** [comment]
> I think making new version of the reactor controller derived from the old one is trivial, and if there is an actual user apperared in the issue, we may safely assume there are more silent ones somewhere out there. I don't think it would be hard for the PR author to do such refactoring and it will decrease objections?
> New controller may even use the same texture, etc.

Yea good idea.  @kuba6000 maybe you can implement that what repo suggest?  And btw i am happy if you change stuff make it more usable to nuild as before.

**Comment by Prometheus0000** [comment]
The discussion is going on in #github-discussion right now. It's better suited to quick back and forths anyway.

**Comment by Prometheus0000** [comment]
This is unlocked now, try not to have circular arguments so that it needs to be locked again. Also check out the discussion on discord if you missed it.

**Comment by Savallator** [comment]
I am not sure what the current state is - 
The HTGR has got seperate cosings now using Europium, is that right? Does it still use the old Casings as base to be able to upgrade the old one? Though i would honestly prefer to even use the same casings and only make the controller different, since that way upgrading the reactor would work without wrenching 400 casings, upgrading them and replacing them. But of course, that would make the whole thing a lot cheaper, not sure if that is desirable.

**Comment by Pxx500** [comment]
There are 2 different controllers now and both of them have their casings. There will be wrenching only if someone wants to make first reactor, as previously said no one does that anyway so it's only your problem and no we won't change that.

**Comment by Savallator** [comment]
The point was that maybe more people would if it gets more useful, and it got already way more usefull since Th232 can actually be obtained in the required Quantities. But still i can live with that. Sorry that i was trying to start helpfull discussion, i guess i will just let you do your thing then...

**Comment by Pxx500** [comment]
Your changes are already present in this PR

**Comment by Prometheus0000** [comment]
I think Pxx500 said the first post was updated on discord, try reading it over again I guess.

File: src/main/java/com/github/bartimaeusnek/bartworks/common/tileentities/multis/GT_TileEntity_HTGR.java [hunk 1]
```diff
@@ -0,0 +1,576 @@
...
+                .addInfo("Consumes up to 0.5% of total Fuel Pellets per Operation depending on efficiency")
+                .addInfo("Efficiency is calculated exponentially depending on the amount of pebbles in the internal buffer")
+                .addInfo("Reactor will take 4 000L/s of coolant multiplied by efficiency and by fuel coolant value (check tooltips)")
+                .addInfo("Uses " + GT_Utility.formatNumbers(powerUsage) + " EU/t")
+                .addInfo("One Operation takes 1 hour")
+                .addSeparator()
+                .beginStructureBlock(11, 12, 11, true)
+                .addController("Front bottom center")
+                .addCasingInfo("Europium Reinforced Radiation Proof Casings", 0)
+                .addStructureInfo("Corners and the 2 touching blocks are air (cylindric)")
```
Comment (GlodBlock) [inline src/main/java/com/github/bartimaeusnek/bartworks/common/tileentities/multis/GT_TileEntity_HTGR.java:145]: are the corners and the 2 touching blocks must be air?

Comment (Pxx500) [reply src/main/java/com/github/bartimaeusnek/bartworks/common/tileentities/multis/GT_TileEntity_HTGR.java:145]: They are ignored by the structure

Comment (kuba6000) [reply src/main/java/com/github/bartimaeusnek/bartworks/common/tileentities/multis/GT_TileEntity_HTGR.java:145]: They dont have to be air, they are ignored in the structure. This comment is just to mark that its cylindric.

**Review by GlodBlock — APPROVED** [review]
it looks good now.

**Comment by Steelux8** [comment]
I was curious about these multis, and after reading these comments and the PR, I'm still not sure about the difference between the two. They're both breeder reactors, or do they have different purposes? Are they different tiers, take in different inputs?

**Comment by Pxx500** [comment]
> I was curious about these multis, and after reading these comments and the PR, I'm still not sure about the difference between the two. They're both breeder reactors, or do they have different purposes? Are they different tiers, take in different inputs?

HTGR is better kekw
HTGR converts Th to Lu 1:1 and THTR 1:4 iirc
HTGR has more fuels with different properties, plutonium generates Xe, uranium Kr , thorium Lu
THTR uses 0,5A@IV
HTGR uses 1A@LuV
THTR can generate up to 20x more hot coolant than HTGR
Both reactors require constant supply of coolant:
-THTR doesn't matter how much coolant you give to it, as long as it gets some it will work and convert it to hot coolant (doesn't affect fuel use)
-HTGR has fixed amount of coolant it needs to be supplied with to keep running (4000L/s Th, 6000L/s U, 8000L/s Pu  but those values are multipled by efficiency so they are the maximum values you get with reactor full of fuel)

You also need europium for HTGR so it's late LuV/ZPM

Both reactors use different fuels (THTR has only one kind of fuel), you need to choose the correct one (they are labelled in game)

One HTGR can supply 8L/s of Xe or Kr at full efficiency

HTGR is also more material efficient since it allows for fuel coat recycling

**Comment by Pxx500** [comment]
@Steelux8 Do you need more info?

**Comment by Steelux8** [comment]
That's plenty of info, thanks. If the HTGR is better, I assume the extra coolant on the THTR is a penalty and not a bonus.

**Comment by Pxx500** [comment]
> If the HTGR is better, I assume the extra coolant on the THTR is a penalty and not a bonus.

THTR is supposed to generate energy and HTGR is supposed to be the better breeder.

**Comment by bombcar** [comment]
How do you even find these things

**Comment by Steelux8** [comment]
> THTR is supposed to generate energy and HTGR is supposed to be the better breeder.

This is what made it clear for me. I felt like they were different tiers of the same thing, but they actually focus on different sides of the same operation. I'll consider their effectiveness when I can actually build one.

**Comment by Pxx500** [comment]
> How do you even find these things

I asked kuba6000 on discord, or you can read this pr also you can just wait untill there's a quest(I will ask kuba6000 about it tomorrow)

**Comment by Pxx500** [comment]
> > THTR is supposed to generate energy and HTGR is supposed to be the better breeder.
> 
> This is what made it clear for me. I felt like they were different tiers of the same thing, but they actually focus on different sides of the same operation. I'll consider their effectiveness when I can actually build one.

You can try both since HTGR casings are made out of THTR casings iirc

**Comment by Steelux8** [comment]
I'm still a fair bit away from tungstensteel, unfortunately. There's also the europium need you mentioned above, although the powergen option seems to be the cheapest one.

**Comment by Pxx500** [comment]
> I'm still a fair bit away from tungstensteel, unfortunately. There's also the europium need you mentioned above, although the powergen option seems to be the cheapest one.

It's supposed to close the energy generation gap before fusion but the energy/uranium dust might be a little off (I don't remember comparing THTRs and fluid reactors efficiency)

**Comment by Steelux8** [comment]
> It's supposed to close the energy generation gap before fusion but the energy/uranium dust might be a little off (I don't remember comparing THTRs and fluid reactors efficiency)

Understandable. I'm all in for nuclear fission, since the only nuclear fusion option we have in the pack is really simplistic after the structure is built, albeit it's required and minimizes lag for a long period of time. I'd much prefer to use other more interesting options, so I'm willing to test stuff like this to see what it can do when running properly, in time.
